simSPI/io.py

```diff
@@ -0,0 +1,297 @@
+"""Helper functions for tem.py processing of input and output files."""
+import mrcfile
+import numpy as np
+
+
+def mrc2data(mrc_file):
+    """Return micrograph from an input .mrc file.
+
+    Parameters
+    ----------
+    mrc_file : str
+        File name for .mrc file to turn into micrograph
+    """
+    with mrcfile.open(mrc_file, "r", permissive=True) as mrc:
+        micrograph = mrc.data
+    if len(micrograph.shape) == 2:
+        micrograph = micrograph[np.newaxis, ...]
+    return micrograph
+
+
+def recursively_save_dict_contents_to_group(h5file, path, dic):
+    """Recursively save dictionary contents to group.
+
+    Parameters
+    ----------
+    h5file : File
+        .hdf5 file to write to.
+    path : str
+        Relative path to save dictionary contents.
+    dic : dict
+        Dictionary containing data.
+    """
+    for k, v in dic.items():
+        if isinstance(v, (np.ndarray, np.int64, np.float64, int, float, str, bytes)):
+            h5file[path + k] = v
+        elif isinstance(v, type(None)):
+            h5file[path + k] = str("None")
+        elif isinstance(v, dict):
+            recursively_save_dict_contents_to_group(h5file, path + k + "/", v)
+        else:
+            raise ValueError("Cannot save %s type" % type(v))
+
+
+def fill_parameters_dictionary(
+    mrc_file=None,
+    pdb_file=None,
+    voxel_size=0.1,
+    particle_name="toto",
+    particle_mrcout=None,
+    crd_file=None,
+    sample_dimensions=[1200, 50, 150],
+    beam_params=[300, 1.3, 100, 0],
+    dose=None,
+    optics_params=[81000, 2.7, 2.7, 50, 3.5, 0.1, 1.0, 0, 0],
+    defocus=None,
+    optics_defocout=None,
+    detector_params=[5760, 4092, 5, 32, "yes", 0.5, 0.0, 0.0, 1.0, 0, 0],
+    noise=None,
+    log_file="simulator.log",
+    seed=-1234,
+):
```

Comment:
would it make more sense to have something here that read in a yaml specific file?

right now I have to look for a comment, then find the line of code that does that, and check the index of the list is correct. 

also the default values are in the comments. I'm scrolling up and down a lot to see if there is reasonable values.

I would re-write this to convert a yaml/config file to something that can be used conveniently to write the .inp file for the simulator

if the tem simulator takes the voxel size in um, but usually the pixel size is in angstroms, this could convert the units. and similarly for other parameters